engine/src/scheme.rs

```diff
@@ -702,6 +718,11 @@ impl SchemeBuilder {
         }
     }
 
+    /// Configures the behavior of not equal comparison against a nil value.
+    pub fn set_nil_not_equal_behavior(&mut self, behavior: bool) {
+        self.nil_not_equal_is_false = !behavior;
```

Comment:
nit: `nil_not_equal_is_true` to avoid needing to invert? Or maybe we can even just name the field `nil_not_equal_behavior` too?

Reply:
I chose this approach so that I could keep deriving Default. If the default is true, we'll need to manually implement it.